src/lib/workspace/DownloadTab/index.tsx

```diff
@@ -249,15 +249,26 @@ export default function DownloadTab({
     );
   }, [WDKStudyReleases, downloadServiceStudyReleases]);
 
+  // We can't show (or sometimes even create!) my subset tables for entities with many vars. Assays in mbio often
+  // have >1000 vars. Until we can handle these cases, hide the whole My Subset section if any entites
+  // are 'assay' entities.
+  const showMySubset =
```

Comment:
Thought I'd pass along some feedback I got early on when doing something similarly, even though your implementation is perfectly fine since the `enhancedEntityData` array is always small.

For a large array, it would be better to leverage a method like `find` since it'll stop iterating through the array as soon as it satisfies the provided callback function. Of course, if it cannot find an array entry that satisfies the condition, then it'll have to iterate through the entire array. But a method like `filter` will always iterate through the entire array.

Reply:
Oh clever!! I'll implement that change. Thank you so much!